components/nexus-bootstrap/src/main/java/org/sonatype/nexus/bootstrap/jetty/ConnectorManager.java

```diff
@@ -0,0 +1,197 @@
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.SslConnectionFactory;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+
+/**
+ * Jetty's internal connector manager to manage custom {@link ConnectorConfiguration}s.
+ * <p/>
+ * Note: This class relies on "id" properties in Jetty XML configuration! Hence, the existing beans in Jetty XML
+ * should NOT be renamed, while adding new (with unique, non-clashing) IDs is okay thing to do.
+ */
+public class ConnectorManager
+{
+  private static final String HTTP_CONFIG_ID = "httpConfig";
+
+  private static final String HTTP_CONNECTOR_ID = "httpConnector";
+
+  private static final String SSL_CONTEXT_FACTORY_ID = "sslContextFactory";
+
+  private static final String HTTPS_CONFIG_ID = "httpsConfig";
+
+  private static final String HTTPS_CONNECTOR_ID = "httpsConnector";
+
+  private final Server server;
+
+  private final Map<String, Object> namedBeans;
+
+  private final IdentityHashMap<ConnectorConfiguration, ServerConnector> managedConnectors;
+
+  public ConnectorManager(final Server server,
+                          final Map<String, Object> namedBeans)
+  {
+    this.server = server;
+    this.namedBeans = namedBeans;
+    this.managedConnectors = new IdentityHashMap<>();
+  }
+
+  /**
+   * Adds and starts connector to Jetty based on passed in configuration.
+   */
+  public ServerConnector addConnector(final ConnectorConfiguration connectorConfiguration)
+  {
+    verifyConfiguration(connectorConfiguration.getScheme());
+
+    final HttpConfiguration httpConfiguration =
+        connectorConfiguration.customize(defaultHttpConfiguration(connectorConfiguration.getScheme()));
+    final ServerConnector connectorPrototype =
+        defaultConnector(connectorConfiguration.getScheme());
+    final ServerConnector serverConnector;
+    if (HttpScheme.HTTP == connectorConfiguration.getScheme()) {
+      serverConnector = new ServerConnector(
+          server,
+          connectorPrototype.getAcceptors(),
+          connectorPrototype.getSelectorManager().getSelectorCount(),
+          new InstrumentedConnectionFactory(
+              new HttpConnectionFactory(httpConfiguration)
+          )
+      );
+    }
+    else if (HttpScheme.HTTPS == connectorConfiguration.getScheme()) {
+      final SslContextFactory sslContextFactory = bean(SSL_CONTEXT_FACTORY_ID, SslContextFactory.class);
+      serverConnector = new ServerConnector(
+          server,
+          connectorPrototype.getAcceptors(),
+          connectorPrototype.getSelectorManager().getSelectorCount(),
+          new InstrumentedConnectionFactory(
+              new SslConnectionFactory(sslContextFactory, "http/1.1")
+          ),
+          new HttpConnectionFactory(httpConfiguration)
+      );
+    }
+    else {
+      throw new IllegalArgumentException("Unsupported HTTP Scheme: " + connectorConfiguration.getScheme());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Unsupported HTTP Scheme: " 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)